ethereum/contracts/EtherVault.sol

```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.9;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+// Holds ETH on behalf of ETHApp
+contract EtherVault is Ownable {
+    event Deposit(address account, uint256 amount);
+    event Withdraw(address account, address recipient, uint256 amount);
+
+    // Accepts ETH from the caller.
+    receive() external payable {
+        emit Deposit(msg.sender, msg.value);
```

Comment:
If we want to scan for events for long historical periods, then the events are not redundant. Vaults can be transferred between multiple apps, and so we lose deposit information (the Lock events) every time ownership is transferred.

For auditing and analysis while in production, I'm very keen for vaults to log everything in detail.

Reply:
That's fair but keep in mind that every user will need to pay extra for this on every single tx they make, and it could be a significant extra cost. We can't decide on whether this trade off makes sense or not without evaluating the actual cost difference, so we should do that.

Reply:
My goal for the governance/upgrade project is to optimize for clarity, security, and auditability. Especially as most of our team aren't very experienced with Solidity and the tradeoffs involved with EVM programming.

If our bridge has huge gas issues, it will likely not be due to low-hanging fruit such as event emission. Rather it would be the high level architecture of our contracts and upgradability design. And that's where my responsibility comes in to make sure we have the right design in place that balances those tradeoffs.

We can always remove the events later on if that ends up being our last resort. That's more efficient than spending time now analysing gas costs in detail for event submission.

Reply:
We could aggregate multiple events to infer this data. Such as the `OwnershipTransferred` event that the vault would emit.